error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/IsInstanceUsage.java

```diff
@@ -0,0 +1,87 @@
+package tech.picnic.errorprone.bugpatterns;
+
+import static com.google.errorprone.BugPattern.LinkType.CUSTOM;
+import static com.google.errorprone.BugPattern.SeverityLevel.SUGGESTION;
+import static com.google.errorprone.BugPattern.StandardTags.STYLE;
+import static com.sun.source.tree.Tree.Kind.IDENTIFIER;
+import static com.sun.source.tree.Tree.Kind.INSTANCE_OF;
+import static com.sun.source.tree.Tree.Kind.LAMBDA_EXPRESSION;
+import static com.sun.source.tree.Tree.Kind.MEMBER_SELECT;
+import static com.sun.source.tree.Tree.Kind.METHOD_INVOCATION;
+import static tech.picnic.errorprone.bugpatterns.util.Documentation.BUG_PATTERNS_BASE_URL;
+
+import com.google.auto.service.AutoService;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.LambdaExpressionTreeMatcher;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.fixes.SuggestedFixes;
+import com.google.errorprone.matchers.Description;
+import com.sun.source.tree.IdentifierTree;
+import com.sun.source.tree.InstanceOfTree;
+import com.sun.source.tree.LambdaExpressionTree;
+import com.sun.source.tree.MemberSelectTree;
+import com.sun.source.tree.MethodInvocationTree;
+
+/** A {@link BugChecker} that aligns usages of T.class::isInstance. */
+@AutoService(BugChecker.class)
+@BugPattern(
+    summary = "Use Class::isInstance where possible",
+    link = BUG_PATTERNS_BASE_URL + "IsInstanceUsage",
+    linkType = CUSTOM,
+    severity = SUGGESTION,
+    tags = STYLE)
+public final class IsInstanceUsage extends BugChecker implements LambdaExpressionTreeMatcher {
+  private static final long serialVersionUID = 1L;
+
+  /** Instantiates a new {@link IsInstanceUsage} instance. */
+  public IsInstanceUsage() {}
+
+  @Override
+  public Description matchLambdaExpression(LambdaExpressionTree tree, VisitorState state) {
+    if (ImmutableSet.of(LAMBDA_EXPRESSION, METHOD_INVOCATION).contains(tree.getKind())) {
+      if (INSTANCE_OF == tree.getBody().getKind()) {
+        return constructDescription(
+            tree, state, constructFix(tree, ((InstanceOfTree) tree.getBody()).getType()));
+      } else if (METHOD_INVOCATION == tree.getBody().getKind()) {
+        return treatMethodInvocation(tree, (MethodInvocationTree) tree.getBody(), state);
+      }
```

Comment:
IIUC the `i -> Integer.class.isInstance(i)` case will be flagged and fixed by [MethodReferenceUsage](https://github.com/PicnicSupermarket/error-prone-support/blob/8fa3ff3702f48d77a23bb12401d1ac4a4993f71a/error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/MethodReferenceUsage.java), so it seems we can simplify this check by dropping support for that case :)

Reply:
Oh, that would be awesome :D 
Let me try to extend the test coverage of MethodReferenceUsage then ;)

Reply:
If there is no suitable/equivalent test yet (didn't check): cool!

Reply:
Yep exactly :)

Reply:
Added this test to `MethodReferenceUsage`:
```
            "    // BUG: Diagnostic contains:",
            "    Stream.of(1).filter(v -> Integer.class.isInstance(v));",`
```
And it was actually not flagged 🤔 
I'm wondering if it's because of the `.class.isInstance` 🤔 
But then it indeed feels like it should be `MethodReferenceUsage` that should be extended there.

Reply:
> But then it indeed feels like it should be `MethodReferenceUsage` that should be extended there.

Indeed! There's [a branch](https://github.com/PicnicSupermarket/error-prone-support/compare/sschroevers/MethodReferenceUsageCheck-improvements) with other improvements for that check, so _perhaps_ it works there. If not, I'd add an `XXX` comment to that effect; it's very tricky code, so I caution against going down that rabbit hole. (Though if you want to... I won't stop you ;).)

Reply:
Hahaha, yeah I actually got inspired from this code, and got slightly afraid 😅 
I'll leave that out for now.
That being said, I'll check in the branch 👌 
And then adapt this code.
Thanks for the quick feedback!

Reply:
Not covered in the new branch, will drop an XXX on `MethodReferenceUsage` and adapt the rest!

Reply:
Done, PTAL :)

Reply:
Tnx! On the review list it goes :smile:

Reply:
Thanks :D